### DISCUSSION TITLE
Unable to create a connection for an Azure Redis service. Connection works with redis-py

### DISCUSSION THREAD

**Discussion by PyDataBlog** [aio-libs-abandoned/aioredis-py#1187] [Q&A]
Hi everyone, I am trying to switch from the official redis python package to this package. 

I currently connect using the official package with:

```python
import redis

redis_client = redis.Redis(
    host="abc",
    port=6380,
    password="def",
    ssl=True,
)

redis_client.ping()  # returns True as expected
```

I am currently getting a connection reset by peer when trying to replicate this with `aioredis`:

```python

async def main():
    redis = aioredis.from_url("redis://def@abc.redis.cache.windows.net:6380")
    return await redis.ping()

res = asyncio.run(main())
```

Am I doing something wrong?

**Comment by m-novikov** [comment]
You can try changing connection string to `"rediss://def@abc.redis.cache.windows.net:6380"` note the double `s` in `rediss` as it implies SSL connection.

**Reply by PyDataBlog** [reply]
Unfortunately, I tried `aioredis.from_url("redis://XX.XXX.XX.X", password="def", port=6380)` and it didn't work @m-novikov

**Reply by m-novikov** [reply]
@PyDataBlog Thanks for testing, but in your connection string I see `redis://` and not `rediss://` as I pointed out, `rediss` prefix implies ssl connection. In addition you can try following:
```python
async def main():
    redis = aioredis.Redis(
        host="abc",
        port=6380,
        password="def",
        ssl=True,
    )
    return await redis.ping()

res = asyncio.run(main())
```
Also could you provide traceback to help debugging your problem.

**Reply by PyDataBlog** [reply]
> @PyDataBlog Thanks for testing, but in your connection string I see `redis://` and not `rediss://` as I pointed out, `rediss` prefix implies ssl connection. In addition you can try following:
> 
> ```python
> async def main():
>     redis = aioredis.Redis(
>         host="abc",
>         port=6380,
>         password="def",
>         ssl=True,
>     )
>     return await redis.ping()
> 
> res = asyncio.run(main())
> ```
> 
> Also could you provide traceback to help debugging your problem.

I can confirm this works! Any difference between this connection and the one suggested by the docs?

**Reply by m-novikov** [reply]
Only difference is that arguments you see passed to `Redis` class would be inferred from the connection url string. If connection string starts with `rediss://` **double** `s` **instead of single** it should behave the same as passing `ssl=True`. If you can't connect using the connection string starting with `rediss://` but cat using above code, please file a bug report.

**Reply by PyDataBlog** [reply]
> Only difference is that arguments you see passed to `Redis` class would be inferred from the connection url string. If connection string starts with `rediss://` **double** `s` **instead of single** it should behave the same as passing `ssl=True`. If you can't connect using the connection string starting with `rediss://` but cat using above code, please file a bug report.

I can confirm that the addition of the 's' made the 'from_url' connection work as well <3

**Reply by m-novikov** [reply]
To clarify, ssl or `rediss://` schema enforce encryption of the connection, so third party can't eavesdrop on communication between redis server and your app. Across cloud platforms (e.g. azure) providing redis as a solution it's common to reject connections if they are not SSL encrypted, this is probably what you encountered trying to connect to SSL server, using non-encrypted connection. 

I am not sure if error message could be improved in this case, because it's possible that no explicit reason is provided when connection is reset.

**Comment by RonaldRegan69** [comment]
I am getting the following error using the above.
![image](https://user-images.githubusercontent.com/84187860/177128256-155ee959-c6bf-4a54-a07b-974ae8f798b0.png)

**Reply by RonaldRegan69** [reply]
there was an issue in my host app_settings host variable. Once fixed it worked as expected
